src/applications/vaos/services/appointment/index.js

```diff
@@ -140,3 +150,149 @@ export function getVAAppointmentLocationId(appointment) {
 
   return null;
 }
+
+/**
+ * Returns the location name of a VA appointment
+ *
+ * @export
+ * @param {Object} appointment A FHIR appointment resource
+ * @returns The location name where the VA appointment is located
+ */
+export function getVAAppointmentLocationName(appointment) {
+  return appointment.participant?.find(p =>
+    p.actor.reference?.startsWith('Location'),
+  )?.actor?.display;
+}
+
+/**
+ * Returns the patient phone number of a VA appointment
+ *
+ * @export
+ * @param {Object} appointment A FHIR appointment resource
+ * @returns The patient phone number where the VA appointment is located
+ */
+export function getPatientPhone(appointment) {
+  return appointment.participant
+    .find(p => p?.actor?.reference.includes('Patient'))
+    ?.actor?.telecom?.find(t => t?.system === 'phone')?.value;
+}
+
+/**
+ * Returns the patient email of a VA appointment
+ *
+ * @export
+ * @param {Object} appointment A FHIR appointment resource
+ * @returns The patient email where the VA appointment is located
+ */
+export function getPatientEmail(appointment) {
+  return appointment.participant
+    .find(p => p?.actor?.reference.includes('Patient'))
+    ?.actor?.telecom?.find(t => t?.system === 'email')?.value;
+}
+
+/**
+ * Filter method for future confirmed appointments
+ * @param {Object} appt A FHIR appointment resource
+ */
+export function filterFutureConfirmedAppointments(appt) {
+  const apptDateTime = moment(appt.start);
+
+  return (
+    !appt.vaos.isPastAppointment &&
+    !FUTURE_APPOINTMENTS_HIDDEN_SET.has(appt.description) &&
+    apptDateTime.isValid() &&
+    apptDateTime.isBefore(moment().add(13, 'months'))
+  );
+}
+
+/**
+ * Filter method for past appointments
+ * @param {Object} appt A FHIR appointment resource
+ */
+export function filterPastAppointments(appt, startDate, endDate) {
+  const apptDateTime = moment(appt.start);
+
+  if (
+    appt.vaos.appointmentType === APPOINTMENT_TYPES.vaAppointment &&
+    PAST_APPOINTMENTS_HIDDEN_SET.has(appt.description)
+  ) {
+    return false;
+  }
+
+  return (
+    apptDateTime.isValid() &&
+    apptDateTime.isAfter(startDate) &&
+    apptDateTime.isBefore(endDate)
+  );
+}
+
+/**
+ * Filter method for past appointment requests
+ * @param {Object} request A FHIR appointment resource
+ */
+export function filterRequests(request) {
+  const hasValidDate = request.requestedPeriod.some(period => {
+    const momentStart = moment(period.start);
+    const momentEnd = moment(period.end);
+    return (
+      momentStart.isValid() &&
+      momentEnd.isValid() &&
+      momentStart.isBefore(moment().add(13, 'months'))
```

Comment:
I think we want to replace this check with the check to see if there are any dates after today. The 13 month logic does seem to do anything useful for us, since we should have stopped showing cancelled requests once the last date is in the past.

Reply:
There's an integration test to this effect, but it's passing because of the change to `isPastAppointment` below.

Reply:
Replacing the isBefore 13 month check with isAfter now